### PR TITLE
Reporter web app

### PR DESCRIPTION
**Pull request by misha-codescoop**
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/671)

### CODE REVIEW

File: reporter/src/web-app/js/src/index.css [hunk 1]
```diff
@@ -5,14 +5,41 @@ body {
     font-family: sans-serif;
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/index.css]: Believe this should not go into index.css but App.css as it's App related.

File: reporter/src/web-app/js/src/index.css [hunk 2]
```diff
@@ -5,14 +5,41 @@ body {
...
+.ort-tree-searched-item {
     background: yellow;
 }
 
-.selected-item {
+.ort-tree-selected-item {
     background: orange;
     transition: background 100ms ease-in;
+}
+.ort-tree-wrapper {
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/index.css]: Missing space

File: reporter/src/web-app/js/src/index.css [hunk 3]
```diff
@@ -5,14 +5,41 @@ body {
     font-family: sans-serif;
 }
 
-.oss-tree-tooltip .ant-tooltip-inner {
+.ort-tree-tooltip .ant-tooltip-inner {
     max-width: 500px;
 }
-.searched-item {
+.ort-tree-searched-item {
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/index.css]: Missing space

File: reporter/src/web-app/js/src/index.css (hunk 1 above)
Comment (misha-codescoop) [reply reporter/src/web-app/js/src/index.css]: that should go to another PR I assume as soon as that is not a change done within current PR

File: reporter/src/web-app/js/src/index.css (hunk 3 above)
Comment (misha-codescoop) [reply reporter/src/web-app/js/src/index.css]: corrected

File: reporter/src/web-app/js/src/index.css (hunk 2 above)
Comment (misha-codescoop) [reply reporter/src/web-app/js/src/index.css]: corrected

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 4]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+      }
+  }
+
+  calcSorted = () => {
+      const { search } = this.state;
+      const selectedNodes = search ? Object.keys(this.treeNodes).filter(node => this.treeNodes[node].name.indexOf(search) >= 0) : [];
+      const sorted = selectedNodes.sort((a, b) => {
+          const aIndexes = a.split('-');
+          const bIndexes = b.split('-');
+          let diff = 0;
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Propose to move diff=0 into for-loop on line 84

Comment (misha-codescoop) [reply reporter/src/web-app/js/src/components/TreeView.js]: moved

File: reporter/src/web-app/js/src/index.css (hunk 1 above)
Comment (misha-codescoop) [reply reporter/src/web-app/js/src/index.css]: index.css - that is top level css that should handle html/body/and other stuff outside App
app - that is wrapper for the App component, and the App.css should contain only stuff related to App component and it's children

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 5]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
-        return (
-            <TreeNode
-                key={treeLeaf.id || treeLeaf.name}
-                title={nameWithTooltip}
-            >
-                {treeLeaf.children.map(childLeaf => this.renderTreeNode(childLeaf))}
-            </TreeNode>
-        );
-    }
+  componentDidUpdate(_, prevState) {
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Fix indentation through out TreeView.js to be 4 spaces not 2

Comment (misha-codescoop) [reply reporter/src/web-app/js/src/components/TreeView.js]: +

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 6]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+      const index = selectedIndex + 1 > sorted.length - 1 ? 0 : selectedIndex + 1;
+      this.scrollTo(sorted[index]);
+      this.setState({
+          selectedIndex: index,
+      });
+  }
+
+  render() {
+      const {
+          tree, expandedKeys, autoExpandParent, selectedIndex, sorted,
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Remove last comma after sorted

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 7]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+          selectedIndex: index,
+      });
+  }
+
+  next = () => {
+      const { selectedIndex, sorted } = this.state;
+      const index = selectedIndex + 1 > sorted.length - 1 ? 0 : selectedIndex + 1;
+      this.scrollTo(sorted[index]);
+      this.setState({
+          selectedIndex: index,
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Remove trailing comma

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 8]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+          </TreeNode>
+      );
+  }
+
+  prev = () => {
+      const { selectedIndex, sorted } = this.state;
+      const index = selectedIndex - 1 < 0 ? sorted.length - 1 : selectedIndex - 1;
+      this.scrollTo(sorted[index]);
+      this.setState({
+          selectedIndex: index,
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Remove trailing comma

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 9]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+          </Tooltip>
+      );
+
+      return (
+          <TreeNode
+              key={`${indexElem}-${treeLeaf.id || treeLeaf.name}`}
+              title={nameWithTooltip}
+              ref={ref => this.treeNodes[`${indexElem}`] = {
+                  ref,
+                  name,
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Remove trailing comma

File: reporter/src/web-app/js/src/components/TreeView.js [hunk 10]
```diff
@@ -32,126 +36,229 @@ class TreeView extends React.Component {
...
+          this.scrollTo(sorted[0]);
+          this.setState({
+              sorted,
+          });
+      }
+  }
+
+  scrollTo = (index) => {
+      if (index && this.treeNodes[index].ref) {
+          let el = ReactDOM.findDOMNode(this.treeNodes[index].ref);
```
Comment (tsteenbe) [inline reporter/src/web-app/js/src/components/TreeView.js]: Change el to elem to be consistent with other references to DOM elements

**Comment by tsteenbe** [comment]
Incorporated the changes in this PR in the  035609e7 Refactor of WebApp to use new initial state report data format

**Pull request closed without merging**